### DISCUSSION TITLE
SDK 1.9.0 - SubContractingTerm BT-773-Tender Rule BR-BT-00773-0037 display

### DISCUSSION THREAD

**Discussion by luispozoWS** [OP-TED/eForms-SDK#916] [Q&A] [answered]
Hello,

We have an XSD problem with BT-773-Tender 
![image](https://github.com/OP-TED/eForms-SDK/assets/158502029/22381e8d-5a2e-417b-9bad-3c6435d82fe6)

For notice type 30, BT-773-Tender is mandatory but also forbidden under one condition.
A rule exists for BT-773-Tender => BR-BT-00773-0037 : "'Subcontracting' (BT-773-Tender) is mandatory in notice type '30 – Contract award notice – sectoral directive, standard regime' except when: 'Technical ID of the tender' (OPT-321-Tender) is not filled in"

The schematron rule says this : 
_"context="/*/ext:UBLExtensions/ext:UBLExtension/ext:ExtensionContent/efext:EformsExtension/efac:NoticeResult/efac:LotTender/efac:SubcontractingTerm[efbc:TermCode/@listName='applicability'][$noticeSubType = '30']">
<assert id="BR-BT-00773-0037" role="ERROR" diagnostics="BT-773-Tender" test="count(efbc:TermCode) &gt; 0 or not(../cbc:ID)">rule|text|BR-BT-00773-0037</assert>_

By analyzing the schematron rule, we understand that the field is verified only when the same field is filled. Which means that when the mandatory field is not completed, the verification is not carried out. Error message BR-BT-00773-0037 can therefore never be displayed.
 
The problem with this rule is that we must have a value in this same field to have the error displayed, which is contradictory technically and functionally.

We also have a similar issue for the BT-19-Lot field.

How to handle this ?

**Comment by YvesJo** [comment, marked as answer]
Hi,
As BT-773 conditions the existence of other Subcontracting fields and another context than the parent should be used to evaluate its existence.
When it comes to forbidden rules on BT-773, current context is fine; however for mandatory rules, the grand parent should be used instead. This will get fixed.
I'll have a close look at the BT-19 case

**Reply by YvesJo** [reply]
For BT-19, predicate had to be removed from the parent node and context updated

**Reply by luispozoWS** [reply]
In which version is it removed and updated ?
Best regards

**Reply by YvesJo** [reply]
Hi,
this will get fixed with SDK 1.12. It won't get backported as this would make the rules stricter for the previous version

**Comment by luispozoWS** [comment]
Hello,

We have also a similar problem in notice types 25 to 31 where BT-135-Procedure and BT-136-Procedure must be filled to avoid an XSD error. Schematron rules exist but can not be identified and displayed to the buyer (same problem as described for BT-773-Tender). 

Will this get fixed in another version ?

Best regards

**Reply by YvesJo** [reply]
Rules context for BT-135-Procedure will be updated to the grant parent instead.

**Comment by luispozoWS** [comment]
We have a similar problem with BT-106-procedure in notice type 17 (SDK 1.9) : A schematrons error message does exist for the BT-106-Procedure field “BR-BT-00106-0023” but cannot be shown because the field is checked only when this same field is filled. Which means that when this field which is both mandatory and forbidden under conditions is not filled in, the check is not carried out.
The message can therefore never be shown to the user

**Reply by YvesJo** [reply]
Thanks, this will get fixed in a next SDK release
